open-craft/learning-paths-plugin
# feat: add endpoint to expose LearningPaths as course-discovery Programs

## Description

This PR adds an endpoint to expose LearningPaths as course-discovery Programs.

## Testing instructions

1. Set up devstack
2. Clone this repository to `<devstack root>/src`
3. Start lms and Discovery: `make dev.up.lms+discovery`
4. Enter lms shell: `make dev.shell.lms`
5. Install this plugin: `pip install -e /edx/src/learning-paths-plugin/`
6. Run migrations: ` ./manage.py lms migrate learning_paths`
7. Restart lms: `$ make dev.stop.lms dev.up.lms`
8. Create some learning paths at http://localhost:18000/admin/learning_paths/learningpath/ and associate some LearningPathSteps (courses) with them.
9. Head to Discovery admin [Partner model](http://localhost:18381/admin/core/partner/) and set the `Programs API URL: of the `edX` partner to `http://edx.devstack.lms:18000/api/v1/` so it points to the endpoint added by this PR.
10. Enter the discovery shell  `make discovery-shell`  and run the import command ` ./manage.py refresh_course_metadata`
11. The command must succeed and if you should see the LearningPaths you previously created have been replicated as Programs under http://localhost:18381/admin/course_metadata/program/. 


## Other information
Private ref: BB-8486
